builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -218,7 +229,11 @@ func resourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("content_language", resp.ContentLanguage)
 	d.Set("content_type", resp.ContentType)
 	d.Set("version_id", resp.VersionId)
-	d.Set("kms_key_id", resp.SSEKMSKeyId)
+	if _, ok := d.GetOk("server_side_encryption"); ok {
+		d.Set("server_side_encryption", resp.ServerSideEncryption)
+	} else if _, ok := d.GetOk("kms_key_id"); ok {
+		d.Set("kms_key_id", resp.SSEKMSKeyId)
```

Comment:
While this seems to be fixing the acceptance test which was previously failing 👍 , there's still one more edge case most resources always try to cover and we should cover here too.

If things change outside of Terraform (either by hand or via a different tool/script), Terraform should be able to detect such changes and pull down what's the real key ID in order to present it to the user in the diff. At this point your implementation would cause TF to ignore these changes.

Here comes the tricky part - if you don't specify `SSEKMSKeyId` attribute in the `Put` API request, AWS will find the default master S3 KMS key and then return that key ID back - i.e. empty field in `Put` != empty field in `Head` 😿 . To avoid `Computed` `kms_key_id` (because that would suppress the diff in certain cases) we could work around that by checking what the default Key ID for a given region is and then compare it here.

e.g.
```go
// Find out whether the default S3 KMS key is in use
out, err := kmsconn.DescribeKey(&kms.DescribeKeyInput{
	KeyId: aws.String("alias/aws/s3"),
})
if err != nil {
	return fmt.Errorf("Failed to describe default S3 KMS key (alias/aws/s3): %s", err)
}

// Only set non-default KMS key ID (one that doesn't match default)
if resp.SSEKMSKeyId != nil && *resp.SSEKMSKeyId != *out.KeyMetadata.Arn {
	log.Printf("[DEBUG] S3 object is encrypted using a non-default KMS Key ID: %s", *resp.SSEKMSKeyId)
	d.Set("kms_key_id", resp.SSEKMSKeyId)
}
```

The alias unfortunately isn't very well documented, but from what I can see it's always available - even if it's new  brand new AWS account - It's created for the first time you try interacting with it.